trading_calendars/exchange_calendar_xdub.py

```diff
@@ -0,0 +1,157 @@
+#
+# Copyright 2018 Quantopian, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from datetime import time
+from pandas import Timestamp
+from pandas.tseries.holiday import (
+    DateOffset,
+    EasterMonday,
+    GoodFriday,
+    Holiday,
+    MO,
+    previous_friday,
+    weekend_to_monday,
+)
+from pytz import timezone
+
+from .common_holidays import (
+    new_years_day,
+    european_labour_day,
+    christmas,
+    weekend_christmas,
+    boxing_day,
+    weekend_boxing_day,
+)
+from .trading_calendar import HolidayCalendar, TradingCalendar
+
+NewYearsDay = new_years_day(observance=weekend_to_monday)
+
+StPatricksDay = Holiday(
+    "St. Patrick's Day",
+    month=3,
+    day=17,
+    end_date='2001',
+    observance=weekend_to_monday,
+)
+
+LabourDay = european_labour_day(end_date='2010', observance=weekend_to_monday)
+
+MayBankHoliday = Holiday(
+    'May Bank Holiday',
+    month=5,
+    day=1,
+    offset=DateOffset(weekday=MO(1)),
+)
+JuneBankHoliday = Holiday(
+    'June Bank Holiday',
+    month=6,
+    day=1,
+    offset=DateOffset(weekday=MO(1)),
+)
+
+LastTradingDayBeforeChristmas = Holiday(
+    'Last Trading Day Before Christmas',
+    month=12,
+    day=24,
+    start_date='2010',
+    observance=previous_friday,
+)
+Christmas = christmas()
+WeekendChristmas = weekend_christmas()
+BoxingDay = boxing_day()
+WeekendBoxingDay = weekend_boxing_day()
+
+LastTradingDayOfCalendarYear = Holiday(
```

Comment:
This and `LastTradingDayBeforeChristmas ` are based on http://www.ise.ie/Media/News-and-Events/2016/Irish-Stock-Exchange-Trading-Calendar-for-2017.html.